### PR TITLE
Fix e2e tests failure in action of build and test

### PR DESCRIPTION
**Pull request by tiationg-kho**
**Issue #, if available:**

**Description of changes:**
- Add custom service client when perform e2e test
- Make sure AEMM is up before NTH
- Fix validation in prometheus-metrics-test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tiationg-kho** [comment]
This pattern follows the approach in webhook-http-proxy-test, and webhook-secret-test. 

I think we should be good for only updating these 3 tests (emit-events-test, prometheus-metrics-test, and webhook-test).

File: pkg/ec2metadata/ec2metadata.go [hunk 1]
```diff
@@ -122,6 +122,18 @@ type NodeMetadata struct {
 
 // New constructs an instance of the Service client
 func New(metadataURL string, tries int) *Service {
+	if strings.HasPrefix(metadataURL, "http://amazon-ec2-metadata-mock-service.default.svc.cluster.local") {
```
Comment (LikithaVemulapalli) [inline pkg/ec2metadata/ec2metadata.go:125]: add a comment to clarify it here or either remove it if tests are passing successfully without it

Comment (tiationg-kho) [reply pkg/ec2metadata/ec2metadata.go]: Fixed by removing.

**Review by LikithaVemulapalli — APPROVED** [review]
/lgtm

**Pull request merged**